Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -0,0 +1,127 @@
+                                  options: FindOneAndDeleteOptions? = nil) throws -> Document? {
+        throw MongoError.commandError(message: "Unimplemented command")
+    }
+
+    /**
+     * Finds a single document and updates it, returning either the original or the updated document.
+     * 
+     * - Parameters:
+     *   - filter: `Document` representing the match criteria
+     *   - replacement: a `Document` to replace the found one
+     *   - options: Optional `FindOneAndUpdateOptions` to use when executing the command
+     *
+     * - Returns: Either the original or updated document depending on selected options, 
+     *            or `nil` if there was no match.
+     */
+    @discardableResult
+    public func findOneAndUpdate(filter: Document, update: Document,
+                                 options: FindOneAndUpdateOptions? = nil) throws -> Document? {
+        throw MongoError.commandError(message: "Unimplemented command")
+    }
+}
+
+/// Indicates which document to return in a find and modify operation.
+public enum ReturnDocument {
+    /// Indicates to return the document before the update, replacement, or insert occured.
+    case before
+
+    ///  Indicates to return the document after the update, replacement, or insert occured.
+    case after
+}
+
+/// Options to use when executing a `findOneAndDelete` command on a `MongoCollection`. 
+public struct FindOneAndDeleteOptions {
+    /// Specifies a collation to use.
+    public let collation: Document?
+
+    /// The maximum amount of time to allow the query to run.
+    public let maxTimeMS: Int64?
+
+    /// Limits the fields to return for all matching documents.
+    public let projection: Document?
+
+    /// The order in which to return matching documents.
+    public let sort: Document?
+}
+
+/// Options to use when executing a `findOneAndReplace` command on a `MongoCollection`. 
+public struct FindOneAndReplaceOptions {
+    /// If `true`, allows the write to opt-out of document level validation.
+    public let bypassDocumentValidation: Bool?
+
+    /// Specifies a collation to use.
+    public let collation: Document?
+
+    /// The maximum amount of time to allow the query to run.
+    public let maxTimeMS: Int64?
+
+    /// Limits the fields to return for all matching documents.
+    public let projection: Document?
+
+    /// When `ReturnDocument.After`, returns the replaced or inserted document rather than the original.
+    public let returnDocument: ReturnDocument?
+
+    /// The order in which to return matching documents.
+    public let sort: Document?
+
+    /// When `true`, creates a new document if no document matches the query.
+    public let upsert: Bool?
+}
+
+/// Options to use when executing a `findOneAndUpdate` command on a `MongoCollection`. 
+public struct FindOneAndUpdateOptions {
+    /// A set of filters specifying to which array elements an update should apply.
+    public let arrayFilters: [Document]?
```

Comment:
AFAICT, FindOneAndUpdateOptions is basically FindOneAndReplaceOptions with the addition of this option. Since Swift doesn't allow struct inheritance, are we stuck with copy-pasting everything?

The same thought also occurred while looking at EstimatedDocumentCountOptions, which only contains a common `maxTimeMS` member.